### PR TITLE
Additional Details constructors

### PR DESCRIPTION
**Pull request by oluwasayo**

### CODE REVIEW

File: vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java [hunk 1]
```diff
@@ -38,6 +38,32 @@
     public Details() {
     }
 
+    /**
+     * Initializes a new Details using the provided summary and content.
+     *
+     * @param summary the summary text to set.
```
Comment (tulioag) [inline vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java]: Missing second param in javadoc. Also, would it make sense to add a constructor in which the content is also a String?

File: vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java [hunk 2]
```diff
@@ -38,6 +38,32 @@
...
+     */
+    public Details(String summary, Component content) {
+        setSummaryText(summary);
+        setContent(content);
+    }
+
+    /**
+     * Initializes a new Details using the provided summary and content.
+     *
+     * @param summary the summary component to set.
```
Comment (tulioag) [inline vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java]: Missing second parameter in javadoc.

File: vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java (hunk 1 above)
Comment (oluwasayo) [reply vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java]: Missing param added to Javadoc.
I would consider avoiding a NxN explosion of constructors. What do you feel?

Comment (tulioag) [reply vaadin-details-flow/src/main/java/com/vaadin/flow/component/details/Details.java]: Yes, and probably content as a component is more useful.

**Pull request merged**